workflow-ui/core-android/src/main/java/com/squareup/workflow1/ui/container/DialogStack.kt

```diff
@@ -0,0 +1,146 @@
+package com.squareup.workflow1.ui.container
+
+import android.content.Context
+import android.os.Parcel
+import android.os.Parcelable
+import android.os.Parcelable.Creator
+import android.view.View
+import androidx.lifecycle.LifecycleOwner
+import com.squareup.workflow1.ui.ViewEnvironment
+import com.squareup.workflow1.ui.WorkflowUiExperimentalApi
+import com.squareup.workflow1.ui.androidx.WorkflowLifecycleOwner
+import com.squareup.workflow1.ui.container.DialogRunner.KeyAndBundle
+
+/**
+ * Does the bulk of the work of maintaining a set of [Dialog][android.app.Dialog]s
+ * to reflect lists of [Overlay]. Can be used to create custom [Overlay]-based
+ * layouts if [BodyAndModalsScreen] or the default [View] bound to it are too restrictive.
+ * Provides a [LifecycleOwner] per managed dialog, and view persistence support.
+ */
+@WorkflowUiExperimentalApi
+public class DialogStack(
```

Comment:
`DialogStackContainer` we use the `Container` prefix for these logic classes that deal with interaction between groups of Views bound to Renderings no?

Reply:
`Container` isn't good b/c that's consistently meant "a workflow view that displays other workflow views"

And as @helios175 pointed out, `Stack` isn't good because this is so completely different from `BackStack`.

How about `LayeredDialogHolders`.

Reply:
Oh, but `DialogHolder` is a private type. So just `LayeredDialogs`.